lib/jekyll/page.rb

```diff
@@ -36,22 +35,22 @@ class Page
     # dir  - The String path between the source and the file.
     # name - The String filename of the file.
     def initialize(site, base, dir, name)
-      @site = site
       @base = base
       @dir  = dir
       @name = name
-      @path = site.in_source_dir(base, dir, name)
 
-      process(name)
-      read_yaml(File.join(base, dir), name)
-
-      data.default_proc = proc do |_, key|
-        site.frontmatter_defaults.find(File.join(dir, name), type, key)
-      end
+      full_path = site.in_source_dir(base, dir, name)
+      super(full_path, {
+        :site => site,
+        :collection => site.pages
+      })
 
-      Jekyll::Hooks.trigger :pages, :post_init, self
```

Comment:
I see you removed the hook here, is it added back in elsewhere? The removal of this might cause it's milestone to shift later than you might want because removing this is a major change that will disrupt users.